File: examples/geometry/tests/bindings/test_geometry.swift
import geometry

let ln1 = Line(start: Point(x: 0, y: 0), end: Point(x: 1, y: 2))
let ln2 = Line(start: Point(x: 1, y: 1), end: Point(x: 2, y: 2))

assert(gradient(ln: ln1) == 2.0)
assert(gradient(ln: ln2) == 1.0)

assert(intersection(ln1: ln1, ln2: ln2) == Point(x: 0, y: 0))
assert(intersection(ln1: ln1, ln2: ln1) == nil)

File: examples/geometry/tests/bindings/test_geometry.rb
# frozen_string_literal: true

require 'geometry'

include Geometry

def assert(condition)
  raise 'Assertion failed!' unless condition
end

ln1 = Line.new(Point.new(0.0,0.0), Point.new(1.0,2.0))
ln2 = Line.new(Point.new(1.0,1.0), Point.new(2.0,2.0))

assert Geometry.gradient(ln1) == 2
assert Geometry.gradient(ln2) == 1

assert Geometry.intersection(ln1, ln2) == Point.new(0, 0)
assert Geometry.intersection(ln1, ln1).nil?
